numpy/lib/function_base.py

```diff
@@ -949,63 +928,70 @@ def histogramdd(sample, bins=10, range=None, normed=False, weights=None):
                 "Found bin edge of size <= 0. Did you specify `bins` with"
                 "non-monotonic sequence?")
 
-    nbin = asarray(nbin)
+    if edges_are_uniform:
+        # nbins are the number of bins for each dimension
+        # len(edges_concat) == sum(n + 1 for n in nbins)
+        nbins = np.array([len(e) - 1 for e in edges])
+        edges_concat = np.concatenate(edges)
+        hist = _histogramdd_uniform(sample, nbins, edges_concat, weights)
+    else:
+        nbin = asarray(nbin)
 
-    # Handle empty input.
-    if N == 0:
-        return np.zeros(nbin-2), edges
+        # Handle empty input.
```

Comment:
Might be tidier to package this into `_histogramdd_nonuniform`, if possible

Reply:
I would rather make that a separate effort/pull request. I intend to investigate/profile nonuniform binning to see if a speed up can be gained over using bincount as is done here.

Reply:
I'm not saying you should write that function in c - just break the current python code out into a sub function, so it reads vaguely like:


```python
if uniform:
    return _histogramdd_uniform(...)
else:
    return _histogramdd_nonuniform(...)
```

Reply:
understood. Though I'd still like to defer this to a following effort.